src/api/providers/litellm.ts

```diff
@@ -99,14 +99,23 @@ export class LiteLlmHandler implements ApiHandler {
 			return message
 		})
 
-		const stream = await this.client.chat.completions.create({
+		const requestPayload: OpenAI.Chat.Completions.ChatCompletionCreateParamsStreaming = {
 			model: this.options.liteLlmModelId || liteLlmDefaultModelId,
 			messages: [enhancedSystemMessage, ...enhancedMessages],
 			temperature,
 			stream: true,
 			stream_options: { include_usage: true },
 			...(thinkingConfig && { thinking: thinkingConfig }), // Add thinking configuration when applicable
-		})
+		}
+
+		if (this.options.taskId) {
+			// LiteLLM expects metadata as an additional field in the payload.
+			;(requestPayload as any).metadata = {
```

Comment:
The semi-colon here is used due to a type issue.  We should use a more appropriate type than "any" here.